modules/standard/Sets.chpl

```diff
@@ -0,0 +1,700 @@
+    */
+    proc const isDisjoint(const ref other: set(eltType, ?)): bool {
+      _enter();
+
+      var result = true;
+      for x in other do
+        // TODO: Can we insert a break in a for statement?
+        if !_dom.contains(x) then 
+          result = false;
+
+      _leave();
+
+      return result;
+    }
+
+    /*
+      Returns `true` if this set and `other` have at least one element in
+      common.
+
+      :arg other: The set to compare against.
+      :return: Whether or not this set and `other` intersect.
+      :rtype: `bool`
+    */
+    proc const isIntersecting(const ref other: set(eltType, ?)): bool {
+      return !isDisjoint(other);
+    }
+
+    /*
+      Remove the item from this set with a value equal to `x`.
+
+      .. warning::
+
+        Removing an element from this set may invalidate existing references
+        to the elements contained in this set.
+    
+      :arg x: The element to remove.
+
+      :throws IllegalArgumentError: If the list contains no such element.
+    */
+    proc remove(const ref x: eltType) throws {
+      _enter();
+
+      if !_dom.contains(x) {
+        _leave();
+        const msg = "No such element in set: " + x:string;
+        // TODO: Replace with more appropriate error?
+        throw new owned
+          IllegalArgumentError(msg);
+      }
+
+      const dbg = _dom.remove(x);
+
+      _leave();
+    }
+
+    /*
+      Remove the item `x` if it is present in this set, but do nothing if it
+      is not.
+
+      .. warning::
+
+        Discarding an element from this set may invalidate existing references
+        to the elements contained in this set.
+
+      :arg x: The element to discard.
+    */
+    proc discard(const ref x: eltType) {
+      try {
+        remove(x);
+      } catch {}
+    }
+
+    /*
+      Clear the contents of this set.
+
+      .. warning::
+
+        Clearing the contents of this set will invalidate all existing
+        references to the elements contained in this set.
+    */
+    proc clear() {
+      _enter();
+      _dom.clear();
+      _leave();
+    }
+
+    /*
+      Iterate over the elements of this set.
+
+      :yields: A reference to one of the elements contained in this set.
+    */
+    iter these() {
+      for x in _dom.these() do
+        yield x;
+    }
+
+    pragma "no doc"
+    iter these(param tag) where tag == iterKind.standalone {
+      for x in _dom.these(tag) do
+        yield x;
+    }
+
+    pragma "no doc"
+    iter these(param tag) where tag == iterKind.leader {
+      for followThis in _dom.these(tag) do
+        yield followThis;
+    }
+
+    pragma "no doc"
+    iter these(param tag, followThis) where tag == iterKind.follower {
+      for x in _dom.these(tag, followThis) do
+        yield x;
+
+    }
+
+    /*
+      Write the contents of this set to a channel.
+
+      :arg ch: A channel to write to.
+    */
+    proc const writeThis(ch: channel) {
+      _enter();
+
+      var count = 1;
+      ch <~> "[";
```

Comment:
Shouldn't we distinguish how different data structures are printed out? Python sets are printed enclosed in `{}`, as opposed to lists which would be printed in `[]`, for example. This can also help us down the road if we want to implement a `readThis`.

Reply:
I'd also expect sets to print out with `{}`

Reply:
Forgot to point out that `{}` is also the mathematical representation for sets